packages/core/merge.js

```diff
@@ -10,41 +14,40 @@ var getOwnPropertyKeys = require("./get-own-property-keys");
  * @returns {*}
  */
 function mergeOne(dst, src) {
-  if (src === undefined) return dst;
-
-  // According to specification arrays must be concatenated.
-  // Also, the '.concat' creates a new array instance. Overrides the 'dst'.
-  if (isArray(src)) return (isArray(dst) ? dst : []).concat(src);
-
-  // Now deal with non plain 'src' object. 'src' overrides 'dst'
-  // Note that functions are also assigned! We do not deep merge functions.
-  if (!isPlainObject(src)) return src;
+  if (src !== undefined) {
+    // According to specification arrays must be concatenated.
+    // Also, the '.concat' creates a new array instance. Overrides the 'dst'.
+    if (isArray(src)) return isArray(dst) ? [...dst, ...src] : [...src];
 
-  var keys = getOwnPropertyKeys(src);
-  for (var i = 0; i < keys.length; i += 1) {
-    var key = keys[i];
-    var desc = Object.getOwnPropertyDescriptor(src, key);
-    // is this a regular property?
-    if (desc.hasOwnProperty('value')) { // eslint-disable-line
-      // Do not merge properties with the 'undefined' value.
-      if (desc.value === undefined) continue;
-
-      var srcValue = src[key];
-      // Recursive calls to mergeOne() must allow only plain objects or arrays in dst
-      var newDst = isPlainObject(dst[key]) || isArray(srcValue) ? dst[key] : {};
-      // deep merge each property. Recursion!
-      dst[key] = mergeOne(newDst, srcValue);
-    } else { // nope, it looks like a getter/setter
-      Object.defineProperty(dst, key, desc);
+    // Now deal with non plain 'src' object. 'src' overrides 'dst'
+    // Note that functions are also assigned! We do not deep merge functions.
+    if (isPlainObject(src)) {
+      getOwnPropertyKeys(src).forEach((key) => {
```

Comment:
This `merge` function was optimised by me. It is also called from many places. However, frankly speaking, not that many people use `deep...` things. :)

Please, return back the performance optimisations. This is stamp's hot place.

Reply:
Now with `for (... of ...)` inside